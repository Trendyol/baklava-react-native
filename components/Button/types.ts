export type ButtonVariantTypes =
  | 'primary'
  | 'secondary'
  | 'success'
  | 'danger'
  | 'passive';

export type ButtonTypeTypes = 'contained' | 'outline' | 'text';

export type ButtonSizeTypes = 'small' | 'medium' | 'large';
