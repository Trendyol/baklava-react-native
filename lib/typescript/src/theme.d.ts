export declare const space: number[];
export declare const fontSizes: number[];
export declare const radii: {
    none: number;
    small: number;
    normal: number;
    large: number;
    xlarge: number;
};
export declare const colors: {
    white: string;
    black: string;
    transparent: string;
    primaryColor: string;
    secondaryColor: string;
    tertiaryColor: string;
    successColor: string;
    dangerColor: string;
    warningColor: string;
    alternativeColor: string;
    featuredColor: string;
    primaryHover: string;
    secondaryHover: string;
    tertiaryHover: string;
    successHover: string;
    dangerHover: string;
    warningHover: string;
    alternativeHover: string;
    featuredHover: string;
    primaryBackground: string;
    accentBackground: string;
    secondaryBackground: string;
    tertiaryBackground: string;
    successBackground: string;
    alertBackground: string;
    warningBackground: string;
    infoBackground: string;
    featuredCBackground: string;
    borderColor: string;
    contentPrimary: string;
    contentSecondary: string;
    contentTertiary: string;
    contentPassive: string;
    gradientPurple: string[];
    gradientOrange: string[];
};
declare const _default: {
    space: number[];
    fontSizes: number[];
    radii: {
        none: number;
        small: number;
        normal: number;
        large: number;
        xlarge: number;
    };
    colors: {
        white: string;
        black: string;
        transparent: string;
        primaryColor: string;
        secondaryColor: string;
        tertiaryColor: string;
        successColor: string;
        dangerColor: string;
        warningColor: string;
        alternativeColor: string;
        featuredColor: string;
        primaryHover: string;
        secondaryHover: string;
        tertiaryHover: string;
        successHover: string;
        dangerHover: string;
        warningHover: string;
        alternativeHover: string;
        featuredHover: string;
        primaryBackground: string;
        accentBackground: string;
        secondaryBackground: string;
        tertiaryBackground: string;
        successBackground: string;
        alertBackground: string;
        warningBackground: string;
        infoBackground: string;
        featuredCBackground: string;
        borderColor: string;
        contentPrimary: string;
        contentSecondary: string;
        contentTertiary: string;
        contentPassive: string;
        gradientPurple: string[];
        gradientOrange: string[];
    };
};
export default _default;
