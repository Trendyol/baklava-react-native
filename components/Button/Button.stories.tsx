import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Button from './Button';
import Box from '../Box/Box';
import InfoIco from '../../src/icons/Info';
import Text from '../Text/Text';
import { ButtonSizeTypes, ButtonTypeTypes, ButtonVariantTypes } from './types';

const variantList = ['primary', 'secondary', 'success', 'danger', 'passive'];
const typeList = ['contained', 'outline', 'text'];
const sizeList = ['small', 'medium', 'large'];

const ButtonMeta: ComponentMeta<typeof Button> = {
  title: 'Button',
  component: Button,
  argTypes: {
    onPress: { action: 'pressed the button' },
    variant: {
      options: variantList,
      control: { type: 'radios' },
    },
    type: {
      options: typeList,
      control: { type: 'radios' },
    },
    size: {
      options: sizeList,
      control: { type: 'radios' },
    },
  },
  args: {
    variant: variantList[0] as ButtonVariantTypes,
    type: typeList[0] as ButtonTypeTypes,
    size: sizeList[1] as ButtonSizeTypes,
  },
};

export default ButtonMeta;

type ButtonStory = ComponentStory<typeof Button>;

export const Basic: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Button
    </Text>
    <Button
      {...args}
      variant={args.variant}
      type={args.type}
      size={args.size}
      text={`${args.variant} ${args.type} ${args.size} button`}
      m={3}
    />
  </>
);

export const Variants: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Button Variants
    </Text>
    <Box>
      <Button
        {...args}
        variant="primary"
        size="large"
        text="Primary Button"
        m={3}
      />
      <Button
        {...args}
        variant="secondary"
        size="large"
        text="Secondary Button"
        m={3}
      />
      <Button
        {...args}
        variant="success"
        size="large"
        text="Success Button"
        m={3}
      />
      <Button
        {...args}
        variant="danger"
        size="large"
        text="Danger Button"
        m={3}
      />
      <Button
        {...args}
        variant="passive"
        size="large"
        text="Passive Button"
        disabled={true}
        m={3}
      />
    </Box>
  </>
);

export const Types: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Button Types
    </Text>
    <Box>
      <Button
        {...args}
        variant="primary"
        size="large"
        text="Contained Button"
        m={3}
      />
      <Button
        {...args}
        variant="primary"
        size="large"
        type="outline"
        text="Outlined Button"
        m={3}
      />
      <Button
        {...args}
        variant="primary"
        size="large"
        type="text"
        text="Text Button"
        m={3}
      />
    </Box>
  </>
);

export const Sizes: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Button Sizes
    </Text>
    <Box>
      <Button
        {...args}
        variant="primary"
        size="large"
        text="Primary Button"
        m={3}
      />
      <Button
        {...args}
        variant="primary"
        size="medium"
        text="Primary Button"
        m={3}
      />
      <Button
        {...args}
        variant="primary"
        size="small"
        text="Primary Button"
        m={3}
        mb={6}
      />
      <Button
        {...args}
        variant="primary"
        size="large"
        text="Primary Button"
        filled={true}
        m={3}
      />
      <Button
        {...args}
        variant="primary"
        size="medium"
        text="Primary Button"
        filled={true}
        m={3}
      />
      <Button
        {...args}
        variant="primary"
        size="small"
        text="Primary Button"
        filled={true}
        m={3}
      />
    </Box>
  </>
);

export const ContainedButtons: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Contained Buttons
    </Text>
    <Box>
      <Button
        {...args}
        variant="primary"
        size="large"
        text="Primary Button"
        m={3}
      />
      <Button
        {...args}
        variant="secondary"
        size="large"
        text="Secondary Button"
        m={3}
      />
      <Button
        {...args}
        variant="success"
        size="large"
        text="Success Button"
        m={3}
      />
      <Button
        {...args}
        variant="danger"
        size="large"
        text="Danger Button"
        m={3}
      />
      <Button
        {...args}
        variant="passive"
        size="large"
        text="Passive Button"
        disabled={true}
        m={3}
      />
    </Box>
  </>
);

export const OutlinedButtons: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Outlined Buttons
    </Text>
    <Box>
      <Button
        {...args}
        variant="primary"
        size="large"
        type="outline"
        text="Primary Button"
        m={3}
      />
      <Button
        {...args}
        variant="secondary"
        size="large"
        type="outline"
        text="Secondary Button"
        m={3}
      />
      <Button
        {...args}
        variant="success"
        size="large"
        type="outline"
        text="Success Button"
        m={3}
      />
      <Button
        {...args}
        variant="danger"
        size="large"
        type="outline"
        text="Danger Button"
        m={3}
      />
    </Box>
  </>
);

export const TextButtons: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Text Buttons
    </Text>
    <Box>
      <Button
        {...args}
        variant="primary"
        size="large"
        type="text"
        text="Primary Button"
        m={3}
      />
      <Button
        {...args}
        variant="secondary"
        size="large"
        type="text"
        text="Secondary Button"
        m={3}
      />
      <Button
        {...args}
        variant="success"
        size="large"
        type="text"
        text="Success Button"
        m={3}
      />
      <Button
        {...args}
        variant="danger"
        size="large"
        type="text"
        text="Danger Button"
        m={3}
      />
      <Button
        {...args}
        variant="passive"
        size="large"
        type="text"
        text="Passive Button"
        disabled={true}
        m={3}
      />
    </Box>
  </>
);

export const IconButtons: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Icon Buttons
    </Text>
    {/* @ts-ignore */}
    <Box flexDirection="row" justifyContent="space-around">
      <Box>
        <Button {...args} variant="primary" text="Save" icon={InfoIco} m={3} />
        <Button
          {...args}
          variant="secondary"
          text="Save"
          icon={InfoIco}
          m={3}
        />
        <Button {...args} variant="success" text="Save" icon={InfoIco} m={3} />
        <Button {...args} variant="danger" text="Save" icon={InfoIco} m={3} />
        <Button
          {...args}
          variant="passive"
          text="Save"
          icon={InfoIco}
          disabled={true}
          m={3}
        />
      </Box>
      <Box>
        <Button
          {...args}
          variant="primary"
          text="Save"
          type="text"
          icon={InfoIco}
          m={3}
        />
        <Button
          {...args}
          variant="secondary"
          text="Save"
          type="text"
          icon={InfoIco}
          m={3}
        />
        <Button
          {...args}
          variant="success"
          text="Save"
          type="text"
          icon={InfoIco}
          m={3}
        />
        <Button
          {...args}
          variant="danger"
          text="Save"
          type="text"
          icon={InfoIco}
          m={3}
        />
        <Button
          {...args}
          variant="passive"
          text="Save"
          type="text"
          icon={InfoIco}
          disabled={true}
          m={3}
        />
      </Box>
    </Box>
  </>
);

export const IconOnlyButtons: ButtonStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Icon Only Buttons
    </Text>
    <Box>
      <Button {...args} variant="primary" size="large" icon={InfoIco} m={3} />
      <Button {...args} variant="secondary" size="large" icon={InfoIco} m={3} />
      <Button {...args} variant="success" size="large" icon={InfoIco} m={3} />
      <Button {...args} variant="danger" size="large" icon={InfoIco} m={3} />
      <Button
        {...args}
        variant="passive"
        size="large"
        icon={InfoIco}
        disabled={true}
        m={3}
      />
    </Box>
  </>
);
