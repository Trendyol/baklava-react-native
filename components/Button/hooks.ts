import React from 'react';

export const useIsPressed = () => {
  const [isPressed, setIsPressed] = React.useState<boolean>(false);

  return {
    pressableProps: {
      onPressIn: () => setIsPressed(true),
      onPressOut: () => setIsPressed(false),
    },
    isPressed,
  };
};
