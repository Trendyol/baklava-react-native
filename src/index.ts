export { ThemeProvider } from '@ergenekonyigit/restyle';
export { default as theme } from './theme';

export { default as Box } from './components/Box/Box';
export { default as Button } from './components/Button/Button';
export { default as Icon } from './components/Icon/Icon';
export { default as Text } from './components/Text/Text';
