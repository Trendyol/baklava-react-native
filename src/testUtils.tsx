import * as React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import theme from './theme';

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

export * from '@testing-library/react-native';

export { customRender as render };
