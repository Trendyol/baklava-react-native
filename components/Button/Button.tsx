import React from 'react';
import { Pressable, StyleSheet } from 'react-native';
import styled from 'styled-components/native';
import { flexbox, color, space } from 'styled-system';
import Text from '../Text/Text';
import theme from '../../src/theme';
import { useIsPressed } from './hooks';
import type {
  ButtonSizeTypes,
  ButtonTypeTypes,
  ButtonVariantTypes,
} from './types';
import type { TextVariantTypes } from '../Text/types';
import { IconNameType } from '../Icon/types';
import Icon from '../Icon/Icon';

const BaseButton = styled(Pressable)`
  ${flexbox}
  ${color}
  ${space}
`;

const StyledButton = styled(BaseButton)`
  ${theme.buttonSize}
  ${theme.buttonType}
`;

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  underlineText: {
    textDecorationLine: 'underline',
  },
});

const Button = ({
  type = 'contained',
  size = 'medium',
  variant = 'primary',
  icon,
  text,
  disabled = false,
  filled = false,
  ...rest
}: {
  variant?: ButtonVariantTypes;
  type?: ButtonTypeTypes;
  size?: ButtonSizeTypes;
  icon?: IconNameType | null;
  text?: string;
  disabled?: boolean;
  filled?: boolean;
  [key: string]: any;
}) => {
  const { pressableProps, isPressed } = useIsPressed(rest.isPressed);

  /* istanbul ignore next */
  const onPressIn = () => pressableProps.onPressIn();
  /* istanbul ignore next */
  const onPressOut = () => pressableProps.onPressOut();

  const buttonTypes = theme.buttonStyles.type({
    variant,
    type,
    isPressed,
    disabled,
  });

  const textVariants = {
    large: 'subtitle02Medium',
    medium: 'subtitle03Medium',
    small: 'subtitle03Medium',
  }[size] as TextVariantTypes;

  const typeColor = disabled
    ? theme.colors.contentPassive
    : isPressed && type !== 'text'
    ? theme.colors.white
    : buttonTypes[type].color;

  const textStyle = type === 'text' && !disabled ? styles.underlineText : {};

  return (
    <StyledButton
      //@ts-ignore
      type={type}
      size={size}
      variant={variant}
      text={text}
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      isPressed={isPressed}
      alignSelf={filled ? 'stretch' : 'flex-start'}
      disabled={disabled}
      {...rest}>
      {icon ? (
        <Icon name={icon} size="small" color={typeColor} testID="button-icon" />
      ) : null}
      {text ? (
        <Text
          variant={textVariants}
          color={typeColor}
          style={textStyle}
          marginLeft={icon ? 3 : 0}
          testID="button-text">
          {text}
        </Text>
      ) : null}
    </StyledButton>
  );
};

Button.defaultProps = {
  ...styles.button,
};

export default Button;
