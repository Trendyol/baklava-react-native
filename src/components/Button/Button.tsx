import React from 'react';
import { ColorValue, Pressable } from 'react-native';
import {
  backgroundColor,
  border,
  createRestyleComponent,
  createVariant,
  layout,
  LayoutProps,
  spacing,
  spacingShorthand,
  SpacingProps,
  SpacingShorthandProps,
  VariantProps,
} from '@ergenekonyigit/restyle';
import Text, { TextVariants } from '../Text/Text';
import { useIsPressed } from './hooks';
import { IconNameType } from '../Icon/types';
import Icon from '../Icon/Icon';
import { Theme } from '../../theme';
import Box from '../Box/Box';
import { variantColorSelector } from './utils';

export type ButtonProps = React.ComponentProps<typeof Box> &
  React.ComponentProps<typeof Pressable> & {
    children?: React.ReactNode;
    onPress?: () => void;
    label?: string;
    tintColor?: ColorValue;
    icon?: IconNameType | null;
    disabled?: boolean;
    isPressed?: boolean;
    filled?: boolean;
    size?: VariantProps<Theme, 'buttonSizeVariants'>['variant'];
    kind?: VariantProps<Theme, 'buttonKindVariants'>['variant'];
    variant?: VariantProps<Theme, 'buttonVariants'>['variant'];
  };

const sizeVariant = createVariant<Theme, 'buttonSizeVariants', 'size'>({
  property: 'size',
  themeKey: 'buttonSizeVariants',
});

const kindVariant = createVariant<Theme, 'buttonKindVariants', 'kind'>({
  property: 'kind',
  themeKey: 'buttonKindVariants',
});

const variantVariant = createVariant<Theme, 'buttonVariants', 'variant'>({
  property: 'variant',
  themeKey: 'buttonVariants',
});

const ButtonContainer = createRestyleComponent<
  ButtonProps &
    VariantProps<Theme, 'buttonSizeVariants'> &
    VariantProps<Theme, 'buttonKindVariants'> &
    VariantProps<Theme, 'buttonVariants'>,
  Theme
>(
  [
    layout,
    spacing,
    border,
    backgroundColor,
    sizeVariant,
    kindVariant,
    variantVariant,
  ],
  Box,
);

type PressableProps = React.ComponentProps<typeof Pressable> &
  LayoutProps<Theme> &
  SpacingProps<Theme> &
  SpacingShorthandProps<Theme>;

const PressableContainer = createRestyleComponent<PressableProps, Theme>(
  [layout, spacing, spacingShorthand],
  Pressable,
);

const Button = ({
  size = 'm',
  kind = 'default',
  variant = 'primary',
  icon,
  label,
  tintColor,
  disabled = false,
  filled = false,
  ...rest
}: ButtonProps) => {
  const { pressableProps, isPressed } = useIsPressed(rest.isPressed);

  /* istanbul ignore next */
  const onPressIn = () => pressableProps.onPressIn();
  /* istanbul ignore next */
  const onPressOut = () => pressableProps.onPressOut();

  const buttonHorizontalPadding: SpacingProps<Theme>['paddingHorizontal'] =
    label
      ? (
          {
            s: 'm',
            m: 'xl',
            l: '2xl',
          } as Readonly<
            Record<
              keyof Omit<Theme['buttonSizeVariants'], 'defaults'>,
              keyof Theme['spacing']
            >
          >
        )[size]
      : (
          {
            s: '2xs',
            m: 'xs',
            l: 's',
          } as Readonly<
            Record<
              keyof Omit<Theme['buttonSizeVariants'], 'defaults'>,
              keyof Theme['spacing']
            >
          >
        )[size];

  const textVariant: TextVariants = (
    {
      s: 'subtitle03Medium',
      m: 'subtitle03Medium',
      l: 'subtitle02Medium',
    } as const
  )[size];

  const variantColors = variantColorSelector({
    variant,
    kind,
    isPressed,
    disabled,
  });

  const typeColor = disabled
    ? 'neutralLighter'
    : isPressed && variant !== 'tertiary'
    ? 'neutralFull'
    : variantColors.color;

  const iconView = React.useMemo(() => {
    if (!icon) {
      return null;
    }
    if (variant === 'transparent') {
      return (
        <Icon name={icon} size="s" color={tintColor} testID="button-icon" />
      );
    }
    return <Icon name={icon} size="s" color={typeColor} testID="button-icon" />;
  }, [icon, typeColor, tintColor, variant]);

  const labelView = React.useMemo(() => {
    if (!label) {
      return null;
    }
    return (
      <Text
        variant={textVariant}
        color={typeColor}
        marginLeft={icon ? '2xs' : 'none'}
        testID="button-text">
        {label}
      </Text>
    );
  }, [icon, label, textVariant, typeColor]);

  return (
    <PressableContainer
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      disabled={disabled}
      {...rest}>
      <ButtonContainer
        size={size}
        kind={kind}
        // @ts-ignore
        variant={variant}
        paddingHorizontal={buttonHorizontalPadding}
        flexDirection="row"
        alignItems="center"
        justifyContent="center"
        alignSelf={filled ? 'stretch' : 'flex-start'}
        {...variantColors}>
        {iconView}
        {labelView}
      </ButtonContainer>
    </PressableContainer>
  );
};

export default Button;
