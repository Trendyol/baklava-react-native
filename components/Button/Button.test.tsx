import React from 'react';
import { theme } from '../../src';
import { fireEvent, render } from '../../src/testUtils';
import Button from '../Button/Button';
import { ReactTestInstance } from 'react-test-renderer';

describe('Button', () => {
  test('should render button correctly', () => {
    // when
    const { toJSON } = render(<Button text="testtesttest" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render given text correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" text="testtesttest" />,
    );
    const buttonComponent = getByTestId('button');

    // then
    expect(buttonComponent.props.text).toBe('testtesttest');
  });

  test('should render given variant and color correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" variant="success" />,
    );
    const buttonComponent = getByTestId('button');

    // then
    expect(buttonComponent.props.variant).toBe('success');
    expect(buttonComponent.props.style[0].color).toBe(theme.colors.white);
    expect(buttonComponent.props.style[0].backgroundColor).toBe(
      theme.colors.successColor,
    );
  });

  test('should render given font correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" size="large" text="test" />,
    );
    const buttonComponent = getByTestId('button');

    // then
    const textComponent = buttonComponent.children[0] as ReactTestInstance;
    expect(textComponent.props.variant).toBe('subtitle02Medium');
    expect(textComponent.props.color).toBe(theme.colors.white);
  });

  test('should render icon correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" size="large" text="test" icon="check" />,
    );
    const buttonIcon = getByTestId('button-icon');

    // then
    expect(buttonIcon).toBeTruthy();
  });

  test('should not render icon component', () => {
    // when
    const { queryByTestId } = render(
      <Button testID="button" size="large" text="test" />,
    );
    const buttonIcon = queryByTestId('button-icon');

    // then
    expect(buttonIcon).not.toBeTruthy();
  });

  test('should press button', () => {
    // given
    const onPress = jest.fn();

    // when
    const { getByTestId } = render(
      <Button testID="button" size="large" text="test" onPress={onPress} />,
    );
    const buttonComponent = getByTestId('button');
    fireEvent.press(buttonComponent);

    // then
    expect(onPress).toBeCalledTimes(1);
  });

  test('should render given default values correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" text="testtesttest" />,
    );
    const buttonComponent = getByTestId('button');

    // then
    expect(buttonComponent.props.variant).toBe('primary');
    expect(buttonComponent.props.size).toBe('medium');
    expect(buttonComponent.props.type).toBe('contained');
  });

  test('should render given disabled correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" text="testtesttest" disabled />,
    );
    const buttonComponent = getByTestId('button');
    const textComponent = getByTestId('button-text');

    // then
    expect(buttonComponent.props.accessibilityState.disabled).toBeTruthy();
    expect(textComponent.props.color).toBe(theme.colors.contentPassive);
  });

  test('should text color remain white on contained and outline type correctly', () => {
    // when
    const { getByTestId } = render(
      <Button
        testID="button"
        text="testtesttest"
        type="outline"
        isPressed={true}
        testOnly_pressed
      />,
    );
    const textComponent = getByTestId('button-text');

    // then
    expect(textComponent.props.color).toBe(theme.colors.white);
  });

  test('should render given underlineText correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" text="testtesttest" type="text" />,
    );
    const textComponent = getByTestId('button-text');

    // then
    expect(textComponent.props.style[1].textDecorationLine).toBe('underline');
  });

  test('should render underline when button disabled', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" text="testtesttest" disabled type="text" />,
    );
    const textComponent = getByTestId('button-text');

    // then
    expect(textComponent.props.style[1].textDecorationLine).not.toBe(
      'underline',
    );
  });

  test('should render given filled correctly', () => {
    // when
    const { getByTestId } = render(
      <Button testID="button" text="testtesttest" disabled filled />,
    );
    const buttonComponent = getByTestId('button');

    // then
    expect(buttonComponent.props.alignSelf).toBe('stretch');
  });
});
