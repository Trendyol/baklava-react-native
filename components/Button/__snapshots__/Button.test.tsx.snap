// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  alignSelf="flex-start"
  collapsable={false}
  flexDirection="row"
  focusable={true}
  isPressed={false}
  justifyContent="center"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="medium"
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#F27A1A",
        "borderColor": "#F27A1A",
        "borderRadius": 6,
        "borderWidth": 1,
        "color": "#ffffff",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 24,
        "paddingVertical": 12,
      },
    ]
  }
  text="testtesttest"
  type="contained"
  variant="primary"
>
  <Text
    color="white"
    marginLeft={0}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "marginLeft": 0,
        },
        Object {},
      ]
    }
    testID="button-text"
    variant="subtitle03Medium"
  >
    testtesttest
  </Text>
</View>
`;
